lgc/disassembler/Disassembler.cpp

```diff
@@ -226,66 +288,54 @@ void ObjDisassembler::processSection(ELFSectionRef sectionRef) {
   bool outputting = !(sectFlags & ELF::SHF_EXECINSTR);
   for (;;) {
     // One iteration of disassembling the section.
-    // Sort the symbols. (Stable sort as there may be duplicate addresses.)
-    stable_sort(symbols);
 
     // If AMDGPU, create a symbolizer, giving it the symbols.
     MCSymbolizer *symbolizerPtr = nullptr;
     if (m_objFile->getArch() == Triple::amdgcn) {
       std::unique_ptr<MCRelocationInfo> relInfo(m_target->createMCRelocationInfo(m_tripleName, *m_context));
       if (relInfo) {
-        std::unique_ptr<MCSymbolizer> symbolizer(
-            m_target->createMCSymbolizer(m_tripleName, nullptr, nullptr, &symbols, m_context, std::move(relInfo)));
+        std::unique_ptr<MCSymbolizer> symbolizer(m_target->createMCSymbolizer(
+            m_tripleName, nullptr, nullptr, &symbols.getSymbolizerSymbols(), m_context, std::move(relInfo)));
```

Comment:
nit: could you add inline comments with function argument names to make this more readable?